core/src/Instance/AadInstanceDiscovery.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Identity.Core.OAuth2;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Core.Instance
+{
+    internal static class AadInstanceDiscovery
+    {
+        // Keys are host strings
```

Comment:
>// Keys are host strings [](start = 8, length = 24)

remove comment, it's already in the field signature. Consider however to comment on the class and make it a singleton instead as that reveals the intent (imo) of this class better.

Reply:
switched to singelton